pkg/apis/infra/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,226 @@
+	Strategy   *WorkerNodeStrategy `json:"strategy,omitempty"`
+}
+
+type WorkerNodeStrategy struct {
+	Type          WorkerNodeStrategyType           `json:"type,omitempty"`
+	RollingUpdate *RollingUpdateWorkerNodeStrategy `json:"rollingUpdate,omitempty"`
+}
+
+type WorkerNodeStrategyType string
+
+const (
+	// RollingUpdateWorkerNodeStrategyType replaces old machines by new one using rolling update.
+	RollingUpdateWorkerNodeStrategyType WorkerNodeStrategyType = "RollingUpdate"
+
+	// OnDeleteWorkerNodeStrategyType replaces old machines when the deletion of the asssoicated machines are completed.
+	OnDeleteWorkerNodeStrategyType WorkerNodeStrategyType = "OnDelete"
+)
+
+type RollingUpdateWorkerNodeStrategy struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable during the update.
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+	// MaxSurge is the maximum number of nodes that can be created above the desired number of nodes during the update.
+	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
+	// DeletePolicy defines the policy used to identify nodes to delete when downscaling.
+	// Valid values are "Random", "Newest" and "Oldest".
+	// Defaults to "Newest".
+	// +kubebuilder:validation:Enum=Random;Newest;Oldest
+	// +optional
+	DeletePolicy *string `json:"deletePolicy,omitempty"`
+}
+
+// NodeConfig defines the configuration for the node.
+type NodeConfig struct {
+	MachineConfig    `json:",inline"`
+	KubernetesConfig `json:",inline"`
+}
+
+// MachineConfig defines the configuration for the machine.
+type MachineConfig struct {
+	// Replicas is the number of replicas of the machine.
+	Replicas int `json:"replicas"`
+	// AWS is the configuration for the AWS machine.
+	AWS *AWSMachineConfig `json:"aws,omitempty"`
+}
+
+// KubernetesConfig defines the configuration for the Kubernetes node.
+type KubernetesConfig struct {
+	// Labels is the set of labels to apply to the nodes.
+	Labeles map[string]string `json:"labels,omitempty"`
+	// Taints is the set of taints to apply to the nodes.
+	Taints []corev1.Taint `json:"taints,omitempty"`
+}
+
+// AWSMachineConfig defines the configuration for the AWS machine.
+type AWSMachineConfig struct {
+	// IAMInstanceProfile is the name of the IAM instance profile to use for the instance.
+	IAMInstanceProfile string `json:"iamInstanceProfile"`
+	// InstanceType is the type of instance to use for the instance.
+	InstanceType string `json:"instanceType"`
+	// SSHKeyName is the name of the SSH key to use for the instance.
+	SSHKeyName string `json:"sshKeyName"`
+	// RootVolumeSize is the size of the root volume in GB.
+	RootVolumeSize *int32 `json:"rootVolumeSize,omitempty"`
+}
+
+// PodIdentityConfig defines the configuration for the pod identity.
+type PodIdentityConfig struct {
+	// ServiceAccountIssuer is the URL of the service account issuer.
+	ServiceAccountIssuer string `json:"serviceAccountIssuer"`
+}
+
+type ResourceRef struct {
```

Comment:
reuse the `objectRef`? Here we need some other fields like namespace, group?

Reply:
`ClusterResourceSet` is experimental feature, I do not think it will support more kinds and you should not reference a resource across namespace?